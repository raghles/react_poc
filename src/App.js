import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.scss';
import Dashboard from './dashboard/Dashboard'
import Register from './register/Register'
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";

function App() {
  return (
    <React.Fragment>
      <header role="banner" className="page-header">React POC</header>
      <section role="contentinfo" className="content">
        <Router>
          <div>
            <Switch>
              <Route path="/dashboard">
                <Dashboard />
              </Route>
              <Route path="/">
                <Register />
              </Route>
            </Switch>
          </div>
        </Router>
      </section>
    </React.Fragment>
  );
}

export default App;
