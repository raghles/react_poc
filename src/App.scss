$pad: 1em;

.page-header {
  padding: $pad;
  background-color: lightsteelblue;
  font-size: x-large;
  text-align: center;
}

.content {
  padding: $pad;
}