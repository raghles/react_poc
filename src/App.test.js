import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('Testing App Component', () => {
  const component = shallow(<App/>);
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have header role as banner', () => {
    expect(component.find('header').prop('role')).toEqual("banner")
  });
  it('should have section role as contentinfo', () => {
    expect(component.find('section').prop('role')).toEqual("contentinfo")
  });
});
