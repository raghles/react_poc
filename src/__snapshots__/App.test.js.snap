// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<Fragment>
  <header
    className="page-header"
    role="banner"
  >
    React POC
  </header>
  <section
    className="content"
    role="contentinfo"
  >
    <BrowserRouter>
      <div>
        <Switch>
          <Route
            path="/dashboard"
          >
            <Connect(Dashboard) />
          </Route>
          <Route
            path="/"
          >
            <withRouter(Connect(Register)) />
          </Route>
        </Switch>
      </div>
    </BrowserRouter>
  </section>
</Fragment>
`;

exports[`Testing App Component should render correctly 1`] = `
<Fragment>
  <header
    className="page-header"
    role="banner"
  >
    React POC
  </header>
  <section
    className="content"
    role="contentinfo"
  >
    <BrowserRouter>
      <div>
        <Switch>
          <Route
            path="/dashboard"
          >
            <Connect(Dashboard) />
          </Route>
          <Route
            path="/"
          >
            <withRouter(Connect(Register)) />
          </Route>
        </Switch>
      </div>
    </BrowserRouter>
  </section>
</Fragment>
`;
